src/transaction-runner.coffee

```diff
@@ -217,4 +220,16 @@ class TransactionRunner
         configuration.emitter.emit 'test error', error, test if error
         return callback()
 
+  isMultipart: (requestOptions) =>
+    caseInsensitiveRequestHeaders = {}
+    for key, value of requestOptions.headers
+      caseInsensitiveRequestHeaders[key.toLowerCase()] = value
+    caseInsensitiveRequestHeaders['content-type']?.indexOf("multipart") > -1
+
+  replaceLineFeedInBody: (transaction, requestOptions) =>
+    transaction.request['body'] = transaction.request['body'].replace(/\n/g, '\r\n')
```

Comment:
My only hope is that if you do modify the body during hooks, and you do this ^^^ `replace` by yourself, you won't introduce double `\r\r` into the body.
E.g.: 
1. user has hooks to modify (replace `\n` with `\r\n`) request body before transaction happens
2. we will do this replacement again

So I'll propose to check that the body does not contain any `\r` character. If it does, we don't have to do the replacement by ourself.

cc. @netmilk